.cirrus.yml

```diff
@@ -22,584 +40,532 @@ task:
+      script:
         - dart --enable-asserts ./dev/bots/test.dart
-        - export CIRRUS_CHANGE_MESSAGE=`cat /tmp/cirrus_change_message.txt`
-        - export CIRRUS_COMMIT_MESSAGE=`cat /tmp/cirrus_commit_message.txt`
+
+    - name: hostonly_devicelab_tests-gradle3-linux
       container:
-        cpu: 4
-        memory: 12G
-    - name: integration_tests_gradle1-linux
+        # Empirically, as of October 2019, hostonly_devicelab_tests-gradle3-linux gets OOM-killed
+        # with only the default 4G, and that's after running for an hour.
+        cpu: 2
+        memory: 8G
       env:
-        SHARD: integration_tests
-        SUBSHARD: gradle1
-      test_script:
-        # Unsetting CIRRUS_CHANGE_MESSAGE and CIRRUS_COMMIT_MESSAGE as they
-        # might include non-ASCII characters which makes Gradle crash.
-        # See: https://github.com/flutter/flutter/issues/24935
-        # This is a temporary workaround until we figure how to properly configure
-        # a UTF8 locale on Cirrus (or until the Gradle bug is fixed).
-        # TODO(amirh): Set the locale to UTF8.
-        - echo "$CIRRUS_CHANGE_MESSAGE" > /tmp/cirrus_change_message.txt
-        - echo "$CIRRUS_COMMIT_MESSAGE" > /tmp/cirrus_commit_message.txt
-        - export CIRRUS_CHANGE_MESSAGE=""
-        - export CIRRUS_COMMIT_MESSAGE=""
+        CPU: 2
+      script:
         - dart --enable-asserts ./dev/bots/test.dart
-        - export CIRRUS_CHANGE_MESSAGE=`cat /tmp/cirrus_change_message.txt`
-        - export CIRRUS_COMMIT_MESSAGE=`cat /tmp/cirrus_commit_message.txt`
+
+    - name: hostonly_devicelab_tests-misc-linux
       container:
-        cpu: 4
-        memory: 12G
-    - name: integration_tests_gradle2-linux
+        # Empirically, as of October 2019, the hostonly_devicelab_tests-misc-linux shard took about
+        # 20 minutes when run with 1 CPU and 4G of RAM, just over a quarter of an hour with 2 CPUs
+        # and 4G of RAM.
+        cpu: 2
       env:
-        SHARD: integration_tests
-        SUBSHARD: gradle2
-      test_script:
-        # Unsetting CIRRUS_CHANGE_MESSAGE and CIRRUS_COMMIT_MESSAGE as they
-        # might include non-ASCII characters which makes Gradle crash.
-        # See: https://github.com/flutter/flutter/issues/24935
-        # This is a temporary workaround until we figure how to properly configure
-        # a UTF8 locale on Cirrus (or until the Gradle bug is fixed).
-        # TODO(amirh): Set the locale to UTF8.
-        - echo "$CIRRUS_CHANGE_MESSAGE" > /tmp/cirrus_change_message.txt
-        - echo "$CIRRUS_COMMIT_MESSAGE" > /tmp/cirrus_commit_message.txt
-        - export CIRRUS_CHANGE_MESSAGE=""
-        - export CIRRUS_COMMIT_MESSAGE=""
+        CPU: 2
+      script:
         - dart --enable-asserts ./dev/bots/test.dart
-        - export CIRRUS_CHANGE_MESSAGE=`cat /tmp/cirrus_change_message.txt`
-        - export CIRRUS_COMMIT_MESSAGE=`cat /tmp/cirrus_commit_message.txt`
+
+    # TODO(ianh): name: add_to_app_tests-linux
```

Comment:
GitHub bug for this?

What is this actually asking us to do? Do we have any add_to_app_tests-linux?

Reply:
Presumably eventually we will have a test for add-to-app on Linux, whether it's Android add-to-app or actual Linux add-to-app.